app/controllers/groups_controller.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+class GroupsController < ApplicationController
+  def index
+    redirect_to root_path unless current_user
+    redirect_to group_path(current_user.group_id) if current_user&.student?
+    @groups = Group.all
+  end
+
+  def show
+    redirect_to root_path unless id == current_user.group_id || !current_user.student?
+    @group = Group.find id
```

Comment:
Better to use before_action: 
before_action :find_group, only: [:show]

def before_action
  @group = Group.find(params[:id])
end